lib/connections/service.go

```diff
@@ -90,11 +90,13 @@ func NewService(cfg *config.Wrapper, myID protocol.DeviceID, mdl Model, tlsCfg *
 
 	// The rate variables are in KiB/s in the UI (despite the camel casing
 	// of the name). We multiply by 1024 here to get B/s.
-	if service.cfg.Options().MaxSendKbps > 0 {
-		service.writeRateLimit = ratelimit.NewBucketWithRate(float64(1024*service.cfg.Options().MaxSendKbps), int64(5*1024*service.cfg.Options().MaxSendKbps))
+
+	if maxSendKbps := service.cfg.Options().MaxSendKbps; maxSendKbps > 0 {
```

Comment:
Store options in a variable and reuse it, as it involves making a copy and acquiring a lock